cli/azd/pkg/project/ai_helper.go

```diff
@@ -0,0 +1,713 @@
+package project
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"slices"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/machinelearning/armmachinelearning/v3"
+	"github.com/azure/azure-dev/cli/azd/pkg/account"
+	"github.com/azure/azure-dev/cli/azd/pkg/ai"
+	"github.com/azure/azure-dev/cli/azd/pkg/convert"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/osutil"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+	"github.com/benbjohnson/clock"
+)
+
+const (
+	// AiHubNameEnvVarName is the environment variable name for the Azure AI Hub name
+	AiHubNameEnvVarName = "AZUREAI_HUB_NAME"
+	// AiProjectNameEnvVarName is the environment variable name for the Azure AI project name
+	AiProjectNameEnvVarName = "AZUREAI_PROJECT_NAME"
+	// AiEnvironmentEnvVarName is the environment variable name for the Azure AI environment name
+	AiEnvironmentEnvVarName = "AZUREAI_ENVIRONMENT_NAME"
+	// AiModelEnvVarName is the environment variable name for the Azure AI model name
+	AiModelEnvVarName = "AZUREAI_MODEL_NAME"
+	// AiEndpointEnvVarName is the environment variable name for the Azure AI endpoint name
+	AiEndpointEnvVarName = "AZUREAI_ENDPOINT_NAME"
+	// AiDeploymentEnvVarName is the environment variable name for the Azure AI deployment name
+	AiDeploymentEnvVarName = "AZUREAI_DEPLOYMENT_NAME"
+	// AiFlowEnvVarName is the environment variable name for the Azure AI flow name
+	AiFlowEnvVarName = "AZUREAI_FLOW_NAME"
+)
+
+type AiHelper interface {
```

Comment:
Is the interface meant for testing? or are we expecting multiple ai-helper implementations?

IIRC, we wanted to stop bringing interfaces just to mock during tests and work with structs-types, mocking the required fields

Reply:
There is no intent to have multiple implementations.  The primary goal for the interface is to support better unit tests with mocking so that everything down the chain does not have to be stubbed out.

Reply:
I though we were agree on stop adding interfaces just for the unit tests benefit.  The idea was about adding the mocked component to the fields of the struct

Reply:
Resolving for now per offline team discussions.